Rosie.Harmony/HarmonyService.cs

```diff
@@ -0,0 +1,252 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Rosie.Services;
+using System.Linq;
+using Harmony;
+using Harmony.DeviceWrappers;
+using System.Threading;
+
+namespace Rosie.Harmony
+{
+	public class HarmonyService : IRosieService
+	{
+		public static void LinkerPreserve()
+		{
+		}
+
+		bool _isConnected;
+		bool _isConnecting;
+
+		List<Hub> hubs = new List<Hub>();
+		ILogger<HarmonyService> _logger;
+		IDeviceManager _deviceManager;
+		IServicesManager _serviceManager;
+
+		public HarmonyService(ILoggerFactory loggerFactory, IDeviceManager deviceManager, IServicesManager serviceManager)
+		{
+			if (loggerFactory != null)
+				_logger = loggerFactory.AddConsole(LogLevel.Information).CreateLogger<HarmonyService>();
+			_logger?.LogInformation("Setup Harmony Hub");
+			_logger?.LogInformation(Description);
+			_deviceManager = deviceManager;
+			_serviceManager = serviceManager;
+		}
+		public string Domain => "HARMONY";
+
+		public string Name => "HARMONYSERVICE";
+
+		public string Description => "Integration with Logitech Harmony Hub";
+
+		public string ServiceIdentifier => nameof(HarmonyService);
+
+		public Task Send()
+		{
+			throw new NotImplementedException();
+		}
+
+		public Task Received(object data)
+		{
+			throw new NotImplementedException();
+		}
+
+		public async Task Start()
+		{
+			_logger.LogInformation("Starting");
+
+			await Connect();
+
+			if (_isConnected)
+				await GetActivities();
+		}
+
+		async Task GetActivities()
+		{
+			var existingActivities = await _deviceManager.GetAllDevices();
+
+			// Add all the activities from the hubs as devices
+			foreach (var hub in hubs) {
+
+				if ((hub?.Activities?.Length ?? 0) <= 0)
```

Comment:
maybe we can remove the possibility of hub == null, also initialize activities as an empty collection to avoid this checks